flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/optimize/program/FlinkRuntimeFilterProgram.java

```diff
@@ -325,6 +322,9 @@ private static Optional<BuildSideInfo> findSuitableBuildSide(
         } else if (rel instanceof Join) {
             // try to push the builder to one input of join
             Join join = (Join) rel;
+            if (!isSuitableJoinType(join.getJoinType())) {
+                return Optional.empty();
+            }
```

Comment:
I have some doubt about this logic:
```
if (!(join.getLeft() instanceof Exchange) && !(join.getRight() instanceof Exchange)) {
                return Optional.empty();
            }
```
I think the Join left and right maybe also HashAgg, the HashAgg input is Exchange, so the RuntimeFilter can also be pushdown into HashAgg, we should not stop it here.